feature/platform/transceiver/otg_tests/zr_laser_bias_current_test/zr_laser_bias_current_test.go

```diff
@@ -0,0 +1,219 @@
+	gnmi.Await(t, dut1, gnmi.OC().Interface(dp1.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_UP)
+	transceiverState := gnmi.Get(t, dut1, gnmi.OC().Interface(dp1.Name()).Transceiver().State())
+	// Check if TRANSCEIVER is of type 400ZR
+	if dp1.PMD() != ondatra.PMD400GBASEZR {
+		t.Fatalf("%s Transceiver is not 400ZR its of type: %v", transceiverState, dp1.PMD())
+	}
+	OCcomponent := opticalChannelComponentFromPort(t, dut1, dp1)
+	component1 := gnmi.OC().Component(OCcomponent)
+	p1StreamInstant := samplestream.New(t, dut1, component1.OpticalChannel().LaserBiasCurrent().Instant().State(), 10*time.Second)
+	p1StreamMin := samplestream.New(t, dut1, component1.OpticalChannel().LaserBiasCurrent().Min().State(), 10*time.Second)
+	p1StreamMax := samplestream.New(t, dut1, component1.OpticalChannel().LaserBiasCurrent().Max().State(), 10*time.Second)
+	p1StreamAvg := samplestream.New(t, dut1, component1.OpticalChannel().LaserBiasCurrent().Avg().State(), 10*time.Second)
+	// Disable interface transceiver power off
+	gnmi.Update(t, dut1, gnmi.OC().Component(dp1.Name()).Transceiver().Enabled().Config(), false)
+	verifyLaserBiasCurrent(t, dut1, p1StreamInstant, "laserbiasInstant")
+	verifyLaserBiasCurrent(t, dut1, p1StreamMin, "laserbiasMin")
+	verifyLaserBiasCurrent(t, dut1, p1StreamMax, "laserbiasMax")
+	verifyLaserBiasCurrent(t, dut1, p1StreamAvg, "laserbiasAvg")
+	// Enable interface transceiver power on
+	gnmi.Update(t, dut1, gnmi.OC().Component(dp1.Name()).Transceiver().Enabled().Config(), true)
+	gnmi.Await(t, dut1, gnmi.OC().Interface(dp1.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_UP)
+	verifyLaserBiasCurrent(t, dut1, p1StreamInstant, "laserbiasInstant")
+	verifyLaserBiasCurrent(t, dut1, p1StreamMin, "laserbiasMin")
+	verifyLaserBiasCurrent(t, dut1, p1StreamMax, "laserbiasMax")
+	verifyLaserBiasCurrent(t, dut1, p1StreamAvg, "laserbiasAvg")
+	defer p1StreamInstant.Close()
+	defer p1StreamMin.Close()
+	defer p1StreamMax.Close()
+	defer p1StreamAvg.Close()
+}
+func opticalChannelComponentFromPort(t *testing.T, dut *ondatra.DUTDevice, p *ondatra.Port) string {
+	t.Helper()
+	if deviations.MissingPortToOpticalChannelMapping(dut) {
+		switch dut.Vendor() {
+		case ondatra.ARISTA:
+			transceiverState := gnmi.Get(t, dut, gnmi.OC().Interface(p.Name()).Transceiver().State())
+			return fmt.Sprintf("%s-Optical0", transceiverState)
+		default:
+			t.Fatal("Manual Optical channel name required when deviation missing_port_to_optical_channel_component_mapping applied.")
+		}
+	}
+	compName := gnmi.Get(t, dut, gnmi.OC().Interface(p.Name()).HardwarePort().State())
+	for {
+		comp, ok := gnmi.Lookup(t, dut, gnmi.OC().Component(compName).State()).Val()
+		if !ok {
+			t.Fatalf("Recursive optical channel lookup failed for port: %s, component %s not found.", p.Name(), compName)
+		}
+		if comp.GetType() == oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_OPTICAL_CHANNEL {
+			return compName
+		}
+		if comp.GetParent() == "" {
```

Comment:
If I follow this correctly, this would check that components that are not optical-channels have a parent. Is this the intent? Perhaps this if statement belongs higher up or as part of the previous if statement?

Reply:
Hi Eric,

As if now we don't have clear mapping from port to opticalchannelcomponent.
we have added a deviation currently.
The above code snippet is to search for any optical channel component recursively for the port.
In the mentioned if condition if it get the component type optical channel it returns it.
if not then it will check if it has parent component.

Reply:
Ah, I see, it recursively goes up the parents. Thanks for explaining.